### PR TITLE
Display action-ref warning

### LINKED ISSUE #2: Emit warning when action is pinned to head (click for more details)
Include this change in all of our actions

https://github.com/google-github-actions/setup-gcloud/pull/460

### PR DESCRIPTION
**Pull request by sameer-coder**
After this change the warning is emitted by the actions-toolkit instead of passing it back to the repo using this toolkit. This way we can get rid of an extra conditional step in the action repo. Tests updated

Closes #2

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
+        `Alternatively, you can pin to any git tag or git SHA in the ` +
+        `repository.`
     )
   }
-  return null
 }
 
 module.exports = {
-  getActionRefWarning
+  displayActionRefWarning
```
Comment (simoneb) [inline src/index.js:28]:
```suggestion
  logActionRefWarning
```

File: test/index.test.js [hunk 2]
```diff
@@ -14,10 +26,12 @@ test('should return warning if actionRef is master', async ({
...
   })
 
+  const { toolkit, warningStub } = setup()
+
   process.env.GITHUB_ACTION_REF = 'master'
-  const warning = getActionRefWarning('nearform/test-repo')
+  toolkit.displayActionRefWarning('nearform/test-repo')
 
-  match(warning, /nearform\/test-repo is pinned at HEAD/)
+  match(warningStub.args[0], /nearform\/test-repo is pinned at HEAD/)
```
Comment (simoneb) [inline test/index.test.js:34]: minor: I prefer using sinon.assert

File: test/index.test.js [hunk 3]
```diff
@@ -48,16 +64,24 @@ test('should return null if actionRef is not main or master', async ({
...
-  equal(warning, null)
+  equal(warningStub.called, false)
 })
 
-test('should return null if invalid repoName', async ({ equal, plan }) => {
+test('should print generic warning if invalid repoName', async ({
+  plan,
+  match
+}) => {
   plan(1)
```
Comment (simoneb) [inline test/index.test.js:79]: no need for planning when you use async

**Review by simoneb — COMMENTED** [review]
see inline comments. please prefer sinon asserts when asserting against mocks

File: test/index.test.js (hunk 2 above)
Comment (sameer-coder) [reply test/index.test.js]: I have changed this to use `calledOnceWithMatch` from sinon which is much better, thanks!

**Review by simoneb — APPROVED** [review]

**Pull request merged**